### PR TITLE
Search page setup

### PR DESCRIPTION
**Pull request by Madhavkapoor28**
*Issue #16934*

*Description of changes:*
Setup for the search page in name-examination. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Can you fix the CI errors?

**Comment by Madhavkapoor28** [comment]
> Can you fix the CI errors?

Done

**Comment by pwei1018** [comment]
Temporary Url for review: https://yfthig-dev--pr-1395-8hhw3uqr.web.app

File: nuxt.config.ts
Comment (severinbeauvais) [inline nuxt.config.ts:18]: Typo I think.

File: package.json
Comment (severinbeauvais) [inline package.json:1]: I recommend adding the following to the top of this file (before line 2):

```
name: "name-examination",
version: "1.0.0",
```

(use appropriate values)

File: nuxt.config.ts
Comment (severinbeauvais) [reply nuxt.config.ts:18]: No, I'm wrong. I guess it really is called fort awesome.

File: package.json
Comment (severinbeauvais) [reply package.json:1]: Then, you can expose the above in the UI somewhere (or dev console), and update it every PR, and easily confirm what version of the app you're using (and whether your change should be there).

File: pages/SearchPage.vue
Comment (severinbeauvais) [inline pages/SearchPage.vue:24]: Why do you both change the icon (arrow-up / arrow-down) and rotate it -- don't you just need to do one of those?

File: pages/SearchPage.vue
Comment (severinbeauvais) [inline pages/SearchPage.vue:165]: Might want to call this `numResults` instead?

File: pages/SearchPage.vue
Comment (severinbeauvais) [inline pages/SearchPage.vue:200]: Wait, so you don't have rich components available to you (like Vuetify)?

Comment (Madhavkapoor28) [reply pages/SearchPage.vue:200]: Unfortunately no, not in tailwind.

Comment (severinbeauvais) [reply pages/SearchPage.vue:200]: How about creating your own (small) component library? You could create a component out of this code, and use it here.

File: package.json
Comment (Madhavkapoor28) [reply package.json:1]: Done

File: pages/SearchPage.vue
Comment (Madhavkapoor28) [reply pages/SearchPage.vue:24]: Yes, removed the arrow-up and made the rotate function 360 deg instead

Comment (severinbeauvais) [reply pages/SearchPage.vue:24]: OK, when I see the new code then maybe it will make sense to me.

But why rotate 360 degrees? Isn't that the same as no rotation?

Comment (Madhavkapoor28) [reply pages/SearchPage.vue:24]: Essentially, yes. When the user clicks on a button, this arrow is rotated 360 deg clockwise with a transition and the dropdown opens if closed. It's rotated 360 counter-clockwise when clicked again to close it.

Comment (severinbeauvais) [reply pages/SearchPage.vue:24]: So it's simply a neat animation?

In other BCROS apps, we use down arrow to indicate a menu open action, and an up arrow to indicate a menu close action, ie:

image

It would be nice to be consistent.

Comment (Madhavkapoor28) [reply pages/SearchPage.vue:24]: Ok, reverting it to rotate-180 makes it consistent. Implemented

Comment (severinbeauvais) [reply pages/SearchPage.vue:24]: Haha, sorry but I appreciate it!

File: components/SearchBox.vue
Comment (ozamani9gh) [inline components/SearchBox.vue:1]: in PROD the word ALL is in CAPS LOCK case, in your code its spelled 'All' please confirm that is not an issue with any code that checks for the word ALL.

File: components/AppHeader.vue
Comment (severinbeauvais) [inline components/AppHeader.vue:191]: Can you use CSS variables (say, in a theme.scss file) and then use them for all the app colours you use?

Comment (severinbeauvais) [reply components/AppHeader.vue:191]: Eg: https://github.com/bcgov/business-filings-ui/blob/main/src/assets/styles/theme.scss

File: components/SearchBox.vue
Comment (severinbeauvais) [inline components/SearchBox.vue:33]: Up to you but I prefer to see the event handler as a function here, eg, `handleFilterChange($event)` -- it looks better / easier to understand in my code editor.

File: package.json
Comment (ozamani9gh) [inline package.json:1]: i am confused about the versioning.  when i check repo, this file is under client folder and it has the following.

"name": "namex",
  "version": "2.1.13",
  "description": "BC Registry: Name Examination",
  
  
  where did you get this package.json.   is this replacing the old one?

File: components/SearchBox.vue
Comment (severinbeauvais) [inline components/SearchBox.vue:156]: Are you going to need to relate these options to specific actions/filters? If so then I recommend using enums for the values, otherwise you're going to end up comparing strings.

File: pages/SearchPage.vue
Comment (severinbeauvais) [reply pages/SearchPage.vue:24]: OK, I looked at the latest code and now it makes sense to me :)
Resolving.

**Review by ozamani9gh — CHANGES_REQUESTED** [review]
quick question about the Dropdown value 'All' compared to PROD 'ALL', will it affect any other part of app, if we lowercase the two LLs.

also package.json currenty is in Client folder with higher version

File: components/SearchBox.vue
Comment (Madhavkapoor28) [reply components/SearchBox.vue:1]: It isn't an issue with any other part of the code, but i've changed it to 'ALL' just to be consistent

File: package.json
Comment (Madhavkapoor28) [reply package.json:1]: This package.json is replacing the old one. The old one (in the master branch) under the client folder, where as in the new name-examination (main branch) it's at the root level.

File: components/SearchBox.vue
Comment (Madhavkapoor28) [reply components/SearchBox.vue:142]: Using enums now

File: components/SearchBox.vue
Comment (severinbeauvais) [reply components/SearchBox.vue]: Thanks, much appreciated! (Easy reviews = good code)

File: components/SearchBox.vue
Comment (severinbeauvais) [reply components/SearchBox.vue:142]: Looks good.

(I did a double-take here, and researched it, but this appears to be the accepted way to get all values from enums.)

File: pages/HomePage.vue
Comment (severinbeauvais) [inline pages/HomePage.vue:53]: Should this be a global setting for eslint (in .eslintrc.js)?

File: pages/HomePage.vue
Comment (severinbeauvais) [inline pages/HomePage.vue:62]: Not critical but I recommend enums for the route names, so that you're not just using strings.

File: pages/SearchPage.vue
Comment (severinbeauvais) [inline pages/SearchPage.vue:195]: Great comment - thanks!

File: store/searchfilters.ts
Comment (severinbeauvais) [inline store/searchfilters.ts:44]: Would it be possible to use the new enum for the key valuese here?

File: components/AppHeader.vue
Comment (severinbeauvais) [inline components/AppHeader.vue:19]: String comparison. Should use an enum :)

**Review by severinbeauvais — COMMENTED** [review]
Looking good so far!

File: pages/HomePage.vue
Comment (Madhavkapoor28) [reply pages/HomePage.vue:62]: using the routes enum now

File: store/searchfilters.ts
Comment (Madhavkapoor28) [reply store/searchfilters.ts:26]: This repo has 3 main pinia stores, 1) auth.ts for the authentication states, 2) fetchstatus.ts, which is used to fetch the stats to be displayed on the landing page, once logged in. And the 3rd once is searchfilters.ts (with class name now renamed to searchFiltersStore) to store the user selected filters and make the API call accordingly. Also, moved helper functions to the bottom

File: components/AppHeader.vue
Comment (Madhavkapoor28) [reply components/AppHeader.vue:19]: Using an enum now

File: pages/HomePage.vue
Comment (Madhavkapoor28) [reply pages/HomePage.vue:53]: Yes it is a bit annoying so i'd prefer to have it as a global rule. Moreover, i just checked, it's disabled in the business-filing-ui repo as well.

Comment (severinbeauvais) [reply pages/HomePage.vue:53]: Do it.

:)

File: store/searchfilters.ts
Comment (severinbeauvais) [reply store/searchfilters.ts:22]: Are `name` or `key` used anywhere else? If so, please consider an enum. If not, go ahead and resolve this conversation.

**Review by severinbeauvais — APPROVED** [review]
Looks good but please wait for another review or 2 before merging.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://yfthig-dev--pr-1395-4gxm77ll.web.app

**Review by pwei1018 — APPROVED** [review]

**Pull request merged**
